addons/arsenal/functions/fnc_scanConfig.sqf

```diff
@@ -282,11 +282,42 @@ uiNamespace setVariable [QGVAR(CBAdisposableLaunchers), compileFinal _launchers]
 uiNamespace setVariable [QGVAR(configItemsTools), compileFinal _toolList];
 
 // Compatibility: Override baseWeapon for RHS optics
-// No good way to do this via script for other attachments, needs manual compat
+// No good way to do this via script for other RHS attachments, needs manual compat
 private _baseWeaponCache = uiNamespace getVariable QGVAR(baseWeaponNameCache);
 {
     private _baseAttachment = configName (_cfgWeapons >> getText (_x >> "rhs_optic_base"));
     if (_baseAttachment != "") then {
-        _baseWeaponCache set [toLower configName _x, _baseAttachment];
+        _baseWeaponCache set [toLowerANSI configName _x, _baseAttachment];
     };
 } forEach ("getText (_x >> 'rhs_optic_base') != ''" configClasses _cfgWeapons);
+
+// Compatibility: Override baseWeapon for CBA Scripted Optics
+// Adapted from https://github.com/Theseus-Aegis/Mods/blob/master/addons/armory/functions/fnc_getBaseVariant.sqf
+private _isScriptedOptic = toString {
+    isClass (_x >> "CBA_ScriptedOptic") ||
+    {(getText (_x >> "weaponInfoType")) regexMatch "CBA_scriptedOptic.*?"}
+};
+
+{
+    private _xClass = toLowerANSI configName _x;
+    private _baseOptic = _xClass call FUNC(baseOptic);
+    if (_baseOptic != "" && {_baseOptic != _xClass}) then {
+        TRACE_2("updating baseOptic",_xClass,_baseOptic);
+        _baseWeaponCache set [_xClass, _baseOptic];
+    };
+} forEach (_isScriptedOptic configClasses _cfgWeapons);
+
+// Compatibility: Override baseWeapon for CBA Scripted Attachments
+private _isScriptedAttachment = toString {
+    isText (_x >> "MRT_SwitchItemNextClass") ||
+    {isText (_x >> "MRT_SwitchItemPrevClass")}
```

Comment:
Some of the RHS attachments have `"MRT_SwitchItemNextClass"` set to `""`, because of inheritance.
I don't think it breaks anything, but it's unnecessary checks.

Reply:
Suggestions?

Reply:
Just went with `getText(...) != ""`.